### DISCUSSION TITLE
Error about GSL during installation on Ubuntu 20.04

### DISCUSSION THREAD

**Discussion by ngrilli** [neperfepx/neper#396] [Q&A] [answered]
Dear Neper Users,

Has anybody encountered the error:
```
/home/nicolo/projects/neper-4.2.0/src/contrib/orilib/../ut/src/ut_array/ut_array.h:10:9: fatal error: gsl/gsl_rng.h: No such file or directory
   10 | #include<gsl/gsl_rng.h>
      |         ^~~~~~~~~~~~~~~
```
when running "make" to compile neper?

I have installed GSL on Ubuntu 20.04 using: 
sudo apt-get install libgsl-dev

I have a conda environment that I tried to deactivate,
but still the same problem appears.

Thank you very much in advance for any suggestion,
Best Regards,
Nicolò

**Comment by rquey** [comment]
I'm not exactly sure what the problem is.

I would have installed both `libgsl23` and `libgsl-dev`, but I don't know if it makes a difference (or if `libgsl23` is a dependency of `libgsl-dev` and got installed).

Did you (re)run `cmake ..` _after_ installing the GSL? What do you see in `ccmake ..`?  Here is what I have:
```
 GSL_CONFIG                       /usr/bin/gsl-config                          
 GSL_CONFIG_PREFER_PATH           /bin;;/bin;                                  
 GSL_EXE_LINKER_FLAGS             -Wl,-rpath,/usr/lib/x86_64-linux-gnu 
```

**Comment by ngrilli** [comment, marked as answer]
Dear @rquey 

Thank you very much for your hints.
The problem is now fixed, the issue was the following:

I had a conda/mamba environment for MOOSE framework,
I deactivated the environment by commenting the relevant part of my bashrc,
so when I open the shell the environment is not loaded.

At this point, without the conda environment I was missing g++, so I did:

sudo apt install g++

then cancelled the build folder and did the procedure again.

cmake ..
make
sudo make install

Hope this can help others,
Best Regards,
Nicolò
